receiver/postgresqlreceiver/client.go

```diff
@@ -163,6 +164,44 @@ func (c *postgreSQLClient) getDatabaseStats(ctx context.Context, databases []str
 	return dbStats, errs
 }
 
+type databaseLocks struct {
+	relation string
+	mode     string
+	lockType string
+	locks    int64
+}
+
+func (c *postgreSQLClient) getDatabaseLocks(ctx context.Context) ([]databaseLocks, error) {
+	query := `SELECT relname AS relation, mode, locktype,COUNT(pid)
+	AS locks FROM pg_locks
+	JOIN pg_class ON pg_locks.relation = pg_class.oid
+	GROUP BY relname, mode, locktype;`
+
+	rows, err := c.client.QueryContext(ctx, query)
+	if err != nil {
+		return nil, fmt.Errorf("unable to query pg_locks and pg_locks.relation: %w", err)
+	}
+	defer rows.Close()
+	var dl []databaseLocks
+	var errors error
```

Comment:
```suggestion
	var errs error
```
to avoid using the same name as the `errors` package.